sunpy/data/_sample.py

```diff
@@ -77,41 +106,38 @@ def download_sample_data(progress=True, overwrite=True, timeout=None):
-    sampledata_dir = get_and_create_sample_dir()
 
-    number_of_files_fetched = 0
-    print("Downloading sample files to {}".format(sampledata_dir))
-    for file_name in six.itervalues(_files):
-        if not overwrite:
-            if os.path.isfile(os.path.join(sampledata_dir,
-                                           file_name[0])):
-                number_of_files_fetched += 1
-                continue
-
-        for base_url in _base_urls:
-            full_file_name = file_name[0] + file_name[1]
+    if filename[-3:] == 'zip':
+        uncompressed_filename = filename[:-4]
+    else:
+        uncompressed_filename = filename
+    #print(uncompressed_filename)
+    # check if the (uncompressed) file exists
+    # print("Downloading sample files to {}".format(sampledata_dir))
+    if not overwrite and os.path.isfile(os.path.join(sampledata_dir, uncompressed_filename)):
+        print("File {} found and overwrite flag not set so skipping.".format(uncompressed_filename))
+    else:
+        # check each provided url to find the file
+        for base_url in url_list:
+            if base_url.count('github'):
+                filename += '?raw=true'
             try:
-                exists = url_exists(os.path.join(base_url, full_file_name))
+                exists = url_exists(os.path.join(base_url, filename))
                 if exists:
-                    f = download_file(os.path.join(base_url, full_file_name))
-                    real_name, ext = os.path.splitext(full_file_name)
+                    f = download_file(os.path.join(base_url, filename))
+                    real_name, ext = os.path.splitext(f)
 
-                    if file_name[1] == '.zip':
+                    if ext == '.zip':
                         print("Unpacking: {}".format(real_name))
                         with ZipFile(f, 'r') as zip_file:
-                            zip_file.extract(real_name, sampledata_dir)
+                            unzipped_f = zip_file.extract(real_name, sampledata_dir)
                         os.remove(f)
+                        move(unzipped_f, os.path.join(sampledata_dir, uncompressed_filename))
                     else:
                         # move files to the data directory
-                        move(f, os.path.join(sampledata_dir, file_name[0]))
+                        move(f, os.path.join(sampledata_dir, uncompressed_filename))
                     # increment the number of files obtained to check later
-                    number_of_files_fetched += 1
                     break
             except (socket.error, socket.timeout) as e:
                 warnings.warn("Download failed with error {}. \n"
                               "Retrying with different mirror.".format(e))
-
-    if number_of_files_fetched < len(list(_files.keys())):
-        raise URLError("Could not download all samples files."
-                       "Problem with accessing sample data servers.")
```

Comment:
This is removed because each file is now dealt with individually. An error will be provided for each file independently so there is no point in issuing another error.